components/form-toggle/index.js

```diff
@@ -21,27 +22,35 @@ class FormToggle extends Component {
 	}
 
 	render() {
-		const { className, checked, children, onChange = noop } = this.props;
-		const id = 'toggle-' + this.id;
+		const {
+			className,
+			checked,
+			onChange = noop,
+			showHint = true,
+			id = 'toggle-' + this.id,
+		} = this.props;
+
 		const wrapperClasses = classNames(
 			'components-form-toggle',
 			className,
 			{ 'is-checked': checked }
 		);
 
 		return (
-			<div className={ wrapperClasses }>
+			<span className={ wrapperClasses }>
 				<input
 					className="components-form-toggle__input"
 					id={ id }
 					type="checkbox"
 					value={ checked }
 					onChange={ onChange }
 				/>
-				<label className="components-form-toggle__label" htmlFor={ id }>
-					{ children }
-				</label>
-			</div>
+				{ showHint &&
+					<span className="components-form-toggle__hint">
```

Comment:
Per #906, it seemed the recommendation would be to disable this from being read by screen readers, I presume because the input's own value state would suffice.

>For this reasons, our decision was to keep using visible "on" and "off" text, hidden to assistive technologies with `aria-hidden="true"` because the state change is already announced by screen readers.

Also discussing the hint more generally at https://github.com/WordPress/gutenberg/issues/906#issuecomment-304297513